### ISSUE TITLE
Using environment variable for square brackets is not documented, not obvious

### ISSUE THREAD

**Issue by matthughes** [elastic/elasticsearch#6352]
The example config in elasticsearch.yml is:

`#discovery.zen.ping.unicast.hosts: ["host1", "host2:port"]`

Using environment variables, I would assume to replace everything in square brackets:
`export ES_HOSTS="[host1, host2:port]"
discovery.zen.ping.unicast.hosts: $ES_HOSTS`

This does not parse correctly though.  The only way I could get it to work was omit the brackets:
`export ES_HOSTS="host1, host2:port"
discovery.zen.ping.unicast.hosts: $ES_HOSTS`

Given that this is not documented, not obvious, and also parses fine as:
`#discovery.zen.ping.unicast.hosts: "host1", "host2:port"`

I propose we change the default config file.  There is also no other use of square brackets in the config file.

**Comment by debadair** [comment]
The relevant topic is now at: https://www.elastic.co/guide/en/elasticsearch/reference/current/settings.html

**Comment by rajat-garg** [comment]
Hey, I want to pick this up as my first OS contribution. Can someone help me?

**Comment by debadair** [comment]
[docs issue triage]

**Comment by lockewritesdocs** [comment]
Looking at the [Configuring Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/7.6/settings.html) page for 7.6, there are no brackets included in examples. The only inclusions of brackets in a supported version are in the 6.8 version, which shows:
```
Enter value for [node.name]:
```

**Comment by lockewritesdocs** [comment]
Talked to @jrodewig about this one, and he thinks that there's value in documenting how arrays work for [environment variables](https://www.elastic.co/guide/en/elasticsearch/reference/current/settings.html#_environment_variable_substitution).

**Comment by lockewritesdocs** [comment]
I did some limited testing with the following variables in `elasticsearch.yml`:
```
node.name: ${HOSTNAME}
discovery.seed_hosts: ${HOST1}, ${HOST2}
```
If I include brackets around the variables `[]` like in an array, Elasticsearch throws an error and fails to start. Entering the variables without the brackets (as shown), I then ran: 
```
export HOST1="127.0.0.1"
export HOST2="127.0.0.1:12"
export HOSTNAME="valhalla"
```
I then started Elasticsearch, which started normally. My takeaway is that brackets are not required when passing multiple variables in `elasticsearch.yml`. I think that we can enhance the examples in the [Environment variable substitution](https://www.elastic.co/guide/en/elasticsearch/reference/current/settings.html#_environment_variable_substitution) section to show multiple variables and indicate that brackets are not required.
